examples/azureml/aml_submit.py

```diff
@@ -0,0 +1,196 @@
+    
+    # create the cluster
+    compute_target = ComputeTarget.create(ws, cluster_name, compute_config)
+    compute_target.wait_for_completion(show_output=True)
+
+#-------------------------------------------------------------------------------
+# Prepare Data
+# Megatron-DeepSpeed takes in data_path, vocab_file, and merge_file.
+# For AML, we are adding a parameter aml_data_download_path which specifies how to deliver the dataset to a compute target.
+# In the submitted run, files in the datasets will be either mounted or downloaded to local path on the compute target.
+# 
+# data_path for this example is path to the .bin and .idx file, excluding extension.
+# e.g. for data/BookCorpusDataset_text_document.bin and data/BookCorpusDataset_text_document.idx,
+# data_path = "data/BookCorpusDataset_text_document"
+#
+# Once the folder is downloaded to the compute target, it will use aml_data_download_path to locate the folder
+# and data_path to locate .bin and .idx files
+#
+# vocab_file and merge_file would also be passed in a similar way.
+#-------------------------------------------------------------------------------
+datastore = ws.get_default_datastore()
+blobstore_datadir = "bookcorpus_data"
+data_path = f"BookCorpusDataset_text_document"
+# Load data folder which contains bookcorpus .bin and .idx files
+train_dataset = Dataset.File.from_files(path=[(datastore, blobstore_datadir)])
+aml_data_download_path = train_dataset.as_download(blobstore_datadir)
+
+vocab_file_dataset = Dataset.File.from_files("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-vocab.json")
+merge_file_dataset = Dataset.File.from_files("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-merges.txt")
+vocab_file = vocab_file_dataset.as_download()
+merge_file = merge_file_dataset.as_download()
+
+
+#-------------------------------------------------------------------------------
+# Setup training environment
+#-------------------------------------------------------------------------------
+megatron_ds_env = Environment.from_dockerfile(name='megatron-ds-ptca', dockerfile='Dockerfile.dockerfile')
+megatron_ds_env.register(ws).build(ws).wait_for_completion()  # Comment this out if environment already exists
+
+megatron_ds_env.environment_variables['NCCL_DEBUG'] = 'WARN'
+megatron_ds_env.environment_variables['CUDA_DEVICE_ORDER'] = 'PCI_BUS_ID'
+megatron_ds_env.environment_variables['NCCL_SOCKET_IFNAME'] = 'eth0'
+megatron_ds_env.environment_variables['NCCL_IB_PCI_RELAXED_ORDERING']='1'
+megatron_ds_env.environment_variables['UCX_TLS']='tcp'
+megatron_ds_env.environment_variables['UCX_NET_DEVICES']='eth0'
+
+#-------------------------------------------------------------------------------
+# Training Settings and Arguments
+#-------------------------------------------------------------------------------
+node_count = 2
+total_proceses_count = 16
```

Comment:
spelling mistake to fix: total_proceses_count --> total_processes or total_process_count